Makefile.include

```diff
@@ -410,6 +410,17 @@ endif
 # Include common programmer logic if available
 -include $(RIOTMAKE)/tools/$(PROGRAMMER).inc.mk
 
+# Check if programmer is supported by the board, only if PROGRAMMERS_SUPPORTED
+# is set and if programmer is not iotlab
+ifneq (iotlab,$(PROGRAMMER))
+  ifneq (,$(PROGRAMMERS_SUPPORTED))
+    ifeq (,$(filter $(PROGRAMMER),$(PROGRAMMERS_SUPPORTED)))
+      $(error '$(PROGRAMMER)' programmer is not supported by this board. \
+              Supported programmers: '$(PROGRAMMERS_SUPPORTED)')
```

Comment:
Maybe we should change this to `info`, I'm thinking of the case where (A) a programmer was missed and (B) I might be testing a new programmer, and maybe it works by magic.

Reply:
Also the good thing about this change is that it wouldn't break anything if we missed anything.

Reply:
I agree and I changed it this way.